Local storage

VeronikaSoshenko opened the pull request:
Сделала локал сторедж

VeronikaSoshenko committed:
verstka

VeronikaSoshenko committed:
Update app.component.html

Удалили поля

VeronikaSoshenko committed:
js

Поправила макет: хэдер корзина, все карточки съехавшие выстроила по размерам, стрелки поставила на место.

VeronikaSoshenko committed:
Дз Js

Исправила в коде мелкие ошибки, выявленные в прошлом код ревью. Перенесла стили прописанные в html в css. Выполнила домашние задания 1-3, 4 тоже, но не понимаю почему оно не работает

VeronikaSoshenko committed:
Роутинг, создание компонента, запрос на сервер

Разделила на отдельные компоненты хидер(новая папка) и мейн боди(оставила в основной папке)

VeronikaSoshenko committed:
resolve-conflicts, add header

VeronikaSoshenko committed:
Merge branch 'routing-2' into routing

VeronikaSoshenko committed:
Merge pull request #1 from Safarov12/routing

Слили ветку с кодом в ветку с картинками

VeronikaSoshenko committed:
Роутинг, создание компонента, запрос на сервер

Сделала задачи по JS, но нужна помощь с 3,8,9. (дз по адресу /hw ) Поработала с юзерами и компонентами

VeronikaSoshenko committed:
Сервисы, классы, передача данных в компонент

Сделала функционал с задачами

VeronikaSoshenko committed:
RXJS

Сделала юзерам сервис и повторила все для задач

VeronikaSoshenko committed:
users and todos form

Исправила ошибки с прошлого дз ( добавила asObservable и протипизировала данные пришедшие через input). Создала формы для юзеров и задачекю

VeronikaSoshenko committed:
Todo form fix

Исправила баг с формой todo для поля выполнения задач. Протипизировала данные из замечения к прошлому дз

VeronikaSoshenko committed:
Убрала any для юзеров

Протипизировала юзеров, создав экспорт интерфейса

VeronikaSoshenko committed:
Вынесла интерфейсы в отдельную папку

Создала папку интерфейсы, в ней два файла с интефейсами для задач и юзеров

VeronikaSoshenko committed:
Подключила ANGULAR MATERIAL

Стилизовала формы юзеров и задачек

VeronikaSoshenko committed:
Дз с модальными окнами

Сделала при нажатии на кнопку “удалить” у карточки пользователя  модальное окно. Сделала верстку карточек пользователя и модальных окон, форм с помощью Angular Material. Добавила [Snackbar]. Итого 3/4 заданий потому что с первым я запуталась в компонентах

VeronikaSoshenko committed:
Пайпы и директивы

Создала пайп многоточия для туду, в хидере вывела сегодняшнюю дату и написала пайп, удаляющий черточки у номера телефона.Добавила директиву для корзины, создала директиву с тенью для юзеров, добавила тултип. Исправила ошибку с мат клоуз и с тем что у меня не редактировались юзеры из-за типизации компани нейм

VeronikaSoshenko committed:
Исправила пайп даты

Исправила пайп даты

VeronikaSoshenko committed:
Guards

Делаю задачу с гвардами, но пока не работает окно ддя входа

VeronikaSoshenko committed:
fix

исправила ошибки, теперь гвард работает

VeronikaSoshenko committed:
LocalStorage task

Сделала задачу с локалстореджем

VeronikaSoshenko committed:
localStorage fix

переписала код через сервисы

pocahontassss commented on src/app/services/local-storage.service.ts:
```diff
@@ -0,0 +1,17 @@
+    return data ? JSON.parse(data) : [];
+  }
+
+  saveUsers(key: string, data: any): void {
+    localStorage.setItem(key, JSON.stringify(data));
+  }
+} 
```
Не протипизирован аргумент. Дженерик в помощь

pocahontassss commented on src/app/services/local-storage.service.ts:
```diff
@@ -0,0 +1,17 @@
+})
+export class StorageService {
+
+  getUsers(key: string): User[] {
+    const data = localStorage.getItem(key);
+    return data ? JSON.parse(data) : [];
+  }
```
Правый операнд должен быть null, иначе внутри сервиса у тебя проверка происходит. Если есть значение по ключу (А пустой массив тоже возвращает true), тогда у тебя всегда будет выполняться условие.

pocahontassss commented on src/app/services/user.service.ts:
```diff
@@ -0,0 +1,60 @@
+    } else {
+      this.usersApiService.getUsers().subscribe(
+        data => {
+          this.localStorage.saveUsers('users', data);
+          this.userSubject$.next(data)
```
Вот эти две строки нужно вынести в отдельный метод setUsers и потом вставлять его во все методы.

VeronikaSoshenko committed:
fix

Протипизировала аргумент, исправила правый операнд на null, создала метод setUsers
